models/classes/action/ActionBlackList.php

```diff
@@ -64,4 +72,28 @@ private function getFeatureFlagChecker(): FeatureFlagChecker
     {
         return $this->getServiceLocator()->get(FeatureFlagChecker::class);
     }
+
+    private function getUserSettingsService(): UserSettingsService
+    {
+        return $this->getServiceManager()->getContainer()->get(UserSettingsService::class);
+    }
+
+    private function isDisabledByInterfaceMode(string $action): bool
+    {
+        if (!in_array($action, self::SIMPLE_INTERFACE_MODE_DISABLED_ACTION)) {
+            return false;
+        }
+
+        $userSettings = $this->getUserSettingsService()->getCurrentUserSettings();
+
+        if (
+            $userSettings->getSetting(
+                UserSettingsInterface::INTERFACE_MODE
+            ) === GenerisRdf::PROPERTY_USER_INTERFACE_MODE_SIMPLE
+        ) {
+            return true;
+        }
+
+        return false;
```

Comment:
chore: can be simplified:

```suggestion
        return $this->getUserSettingsService()
            ->getCurrentUserSettings()
            ->getSetting(UserSettingsInterface::INTERFACE_MODE) === GenerisRdf::PROPERTY_USER_INTERFACE_MODE_SIMPLE
        );
```